app/uk/gov/hmrc/cgtpropertydisposals/service/returns/ReturnsService.scala

```diff
@@ -94,9 +94,10 @@ class DefaultReturnsService @Inject() (
     config.underlying.get[List[CountryCode]]("des.non-iso-country-codes").value
 
   override def submitReturn(
-    returnRequest: SubmitReturnRequest
+    returnRequest: SubmitReturnRequest,
+    representeeDetails: Option[RepresenteeDetails]
```

Comment:
Can you just explain why this method has been extended to add `RepresenteeDetails`? Not 100% with the business logic but that info is already in SubmitReturnRequest, right?

Reply:
I put it in mainly because we need to be able to get a concrete id for the representee to give to DES/ETMP. The `RepresenteeAnswers` has something of type `RepresenteeRefrenceId` in it but one of the values is `NoReferenceId`. In that instance, a call to register without id and to subscribe should be made to get a cgt reference. The intention of `RepresenteeDetails` was to pass in something which would make sure that an id for the representee was definitely available when translating from our case classes to DES data structures